sql/dd/impl/bootstrap/bootstrapper.cc

```diff
@@ -1319,10 +1331,11 @@ bool initialize_dd_properties(THD *thd) {
       unsupported DD_ENGINE; use INNODB if couldn't find DD_ENGINE, since
       existing INNODB DDSE instance doesn't contain DD_ENGINE property
     */
-    uint actual_dd_engine = DB_TYPE_UNKNOWN;
+    legacy_db_type actual_dd_engine = DB_TYPE_UNKNOWN;
     exists = false;
-    if (dd::tables::DD_properties::instance().get(thd, "DD_ENGINE",
-                                                  &actual_dd_engine, &exists) ||
+    if (dd::tables::DD_properties::instance().get(
+            thd, "DD_ENGINE", reinterpret_cast<uint *>(&actual_dd_engine),
```

Comment:
`reinterpret_cast` looks suspicious for an enum (with underlying `int`) to `uint`. Does a `static_cast` result in a compilation error?

Reply:
yep. it complains Static_cast from 'legacy_db_type *' to 'uint *' (aka 'unsigned int *') is not allowed.